primitives/src/lib.rs

```diff
@@ -27,17 +27,35 @@ pub use polkadot_primitives::v1::{
 
 #[cfg(feature = "std")]
 pub mod genesis;
-pub mod xcmp;
+
+/// An inbound HRMP message.
+pub type InboundHrmpMessage = polkadot_primitives::v1::InboundHrmpMessage<relay_chain::BlockNumber>;
+
+/// And outbound HRMP message
+pub type OutboundHrmpMessage = polkadot_primitives::v1::OutboundHrmpMessage<ParaId>;
 
 /// Identifiers and types related to Cumulus Inherents
 pub mod inherents {
 	use sp_inherents::InherentIdentifier;
-
-	/// Inherent identifier for downward messages.
-	pub const DOWNWARD_MESSAGES_IDENTIFIER: InherentIdentifier = *b"cumdownm";
-
-	/// The type of the inherent downward messages.
-	pub type DownwardMessagesType = sp_std::vec::Vec<crate::InboundDownwardMessage>;
+	use sp_std::{
+		vec::Vec,
+		collections::btree_map::BTreeMap,
+	};
+	use super::{InboundDownwardMessage, InboundHrmpMessage, ParaId};
+
+	/// Inherent identifier for message ingestion inherent.
+	pub const MESSAGE_INGESTION_IDENTIFIER: InherentIdentifier = *b"msgingst";
+	/// The data passed via a message ingestion inherent. Consists of a bundle of
+	/// DMP and HRMP messages.
+	#[derive(codec::Encode, codec::Decode, sp_core::RuntimeDebug, Clone, PartialEq)]
+	pub struct MessageIngestionType {
+		/// Downward messages in the order they were sent.
+		pub dmp: Vec<InboundDownwardMessage>,
```

Comment:
What does this `p` stand for?

And honestly I would like to have `downward_messages` and `horizontal_messages`.

I always need to think 5 minutes on what these abbreviations mean. :(

Reply:
gotcha, will deabbreviate